src/textual/_region_group.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from operator import attrgetter
+from typing import NamedTuple, Iterable
+
+from src.textual.geometry import Region
+
+
+class InlineRange(NamedTuple):
+    """Represents a region on a single line."""
+
+    line_index: int
+    start: int
+    end: int
+
+
+def regions_to_ranges(regions: Iterable[Region]) -> Iterable[InlineRange]:
+    """Converts the regions to non-overlapping horizontal strips, where each strip
+    represents the region on a single line. Combining the resulting strips therefore
+    results in a shape identical to the combined original regions.
+
+    Args:
+        regions (Iterable[Region]): An iterable of Regions.
+
+    Returns:
+        Iterable[InlineRange]: Yields InlineRange objects representing the content on
+            a single line, with overlaps removed.
+    """
+    if not regions:
```

Comment:
This may always be True for some iterables.

```python
>>> l=[]
>>> i=iter(l)
>>> bool(i)
True
```

I suspect it is fine to remove this check. The caller will know if it is empty, and if its not it is very little work.

Reply:
TIL!